qiskit/primitives/statevector_sampler.py

```diff
@@ -0,0 +1,225 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023, 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Statevector Sampler class
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Dict, Iterable, List, Optional, Tuple, Union
+
+import numpy as np
+from numpy.typing import NDArray
+
+from qiskit import ClassicalRegister, QiskitError, QuantumCircuit
+from qiskit.circuit import ControlFlowOp
+from qiskit.quantum_info import Statevector
+
+from .base import BaseSamplerV2
+from .base.validation import _has_measure
+from .containers import (
+    BasePrimitiveOptions,
+    BitArray,
+    PrimitiveResult,
+    PubResult,
+    SamplerPub,
+    SamplerPubLike,
+    make_data_bin,
+)
+from .containers.bit_array import _min_num_bytes
+from .primitive_job import PrimitiveJob
+from .utils import bound_circuit_to_instruction
+
+
+@dataclass
+class Options(BasePrimitiveOptions):
+    """Options for execution."""
+
+    shots: int = 100  # TODO: discuss the default number of shots
+    seed: Optional[Union[int, np.random.Generator]] = None
```

Comment:
I tentatively defined this option class. Should I put `shots` and `seed` to a nested `ExecutionOptions`?
Also, what is the good default number of `shots`?

Reply:
For the number of shots, I suggested 512 elsewhere. However, it's more or less free, so if you want to bump it up to 4K or something, feel free. At the end of the day, sampler users who don't bother to set the number of shots are probably just testing something, because if they weren't, they'd probably want to set it.

Reply:
Thanks. I use 512 as default number of shots.